mithril-aggregator/src/services/snapshotter/compressed_archive_snapshotter.rs

```diff
@@ -454,30 +510,109 @@ mod tests {
     }
 
     mod snapshot_ancillary {
-        use mithril_common::digesters::VOLATILE_DIR;
         use mithril_common::test_utils::fake_keys;
 
         use super::*;
 
+        #[test]
+        fn getting_files_to_include_copy_them_to_a_target_directory_while_keeping_source_dir_structure(
+        ) {
+            let test_dir = temp_dir_create!();
+            let cardano_db = DummyCardanoDbBuilder::new(current_function!())
+                .with_immutables(&[1, 2])
+                .with_ledger_files(&["737"])
+                .build();
+            let snapshotter =
+                snapshotter_for_test(&test_dir, cardano_db.get_dir(), CompressionAlgorithm::Gzip);
+            let ancillary_snapshot_dir = test_dir.join("ancillary_snapshot");
+            fs::create_dir(&ancillary_snapshot_dir).unwrap();
+
+            snapshotter
+                .get_files_and_directories_for_ancillary_snapshot(1, &ancillary_snapshot_dir)
+                .unwrap();
+
+            assert_dir_eq!(
+                &ancillary_snapshot_dir,
+                format!(
+                    "* {IMMUTABLE_DIR}/
+                     ** 00002.chunk
+                     ** 00002.primary
+                     ** 00002.secondary
+                     * {LEDGER_DIR}/
+                     ** 737"
+                )
+            );
+        }
+
+        #[tokio::test]
+        async fn delete_temporary_working_directory_after_snapshot_is_created() {
+            let test_dir = temp_dir_create!();
+            let cardano_db = DummyCardanoDbBuilder::new(current_function!())
+                .with_immutables(&[1, 2])
+                .with_ledger_files(&["637"])
+                .build();
+            let snapshotter = CompressedArchiveSnapshotter {
+                ancillary_signer: Arc::new(MockAncillarySigner::that_succeeds_with_signature(
+                    fake_keys::signable_manifest_signature()[0],
+                )),
+                ..snapshotter_for_test(&test_dir, cardano_db.get_dir(), CompressionAlgorithm::Gzip)
+            };
+
+            snapshotter
+                .snapshot_ancillary(1, "ancillary")
+                .await
+                .unwrap();
+
+            let temp_ancillary_snapshot_dir =
+                snapshotter.temp_ancillary_snapshot_directory("ancillary");
+            assert!(
+                !temp_ancillary_snapshot_dir.exists(),
+                "Temporary ancillary snapshot directory has been deleted: {}",
```

Comment:
To enhance clarity, I'd suggest rephrasing the assertion message to something like:
```suggestion
                "Expected temporary ancillary snapshot directory to be deleted, but it still exists: {}",
```